vertical-pod-autoscaler/pkg/admission-controller/logic/limit_range_calculator_test.go

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logic
+
+import (
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/informers"
+
+	//"fmt"
+	"testing"
+
+	//"k8s.io/apimachinery/pkg/runtime"
+	//"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes/fake"
+
+	apiv1 "k8s.io/api/core/v1"
+	//"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/autoscaler/vertical-pod-autoscaler/pkg/utils/test"
+)
+
+func getPod() *apiv1.Pod {
+	labels := map[string]string{"app": "testingApp"}
+	return test.Pod().WithName("test_uninitialized").AddContainer(test.BuildTestContainer("container1", "", "")).WithLabels(labels).Get()
+}
+
+func TestNewNoopLimitsChecker(t *testing.T) {
+	nlc := NewNoopLimitsCalculator()
+	limitRange, err := nlc.GetContainerLimitRangeItem(getPod().Namespace)
+	if assert.NoError(t, err) {
+		assert.Nil(t, limitRange)
+	}
+}
+
+func TestNoLimitRange(t *testing.T) {
+	cs := fake.NewSimpleClientset()
+	factory := informers.NewSharedInformerFactory(cs, 0)
+	lc, err := NewLimitsRangeCalculator(factory)
+
+	if assert.NoError(t, err) {
+		limitRange, err := lc.GetContainerLimitRangeItem(getPod().Namespace)
+		if assert.NoError(t, err) {
+			assert.Nil(t, limitRange)
+		}
+	}
+}
+
+func TestUpdateResourceLimits(t *testing.T) {
+
+	testCases := []struct {
+		name         string
+		pod          *apiv1.Pod
+		limitRanges  []runtime.Object
+		expectErr    error
+		expectLimits *apiv1.LimitRangeItem
+	}{
+		{
+			name: "no matching limit ranges",
+			pod:  getPod(),
+			limitRanges: []runtime.Object{
+				test.LimitRange().WithName("different-namespace").WithNamespace("different").WithType(apiv1.LimitTypePod).WithMax(test.Resources("2", "2")).Get(),
```

Comment:
nit: since it's different namespace already maybe it's not worth changing type too?